### PR TITLE
SCW-346: Send Transaction fix

### PR DESCRIPTION
**Pull request by lluiserena**
## Description

Fixed sendTransaction with the new ZDev version and Viem. 

## Test plan

Send a transaction using the AA SDK.

### CODE REVIEW

**Review by jordiamez — APPROVED** [review]

File: packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts [hunk 1]
```diff
@@ -132,12 +130,39 @@ export class EVMAAWallet<B extends EVMBlockchainIncludingTestnet = EVMBlockchain
...
-            transaction,
-        });
-        throw new TransactionError(`Error sending transaction: In maintenance`);
+        try {
+            const decoratedTransaction = await decorateSendTransactionData(transaction);
+
+            return await this.kernelClient.sendTransaction({
+                to: decoratedTransaction.to as `0x${string}`,
+                value: decoratedTransaction.value ? BigInt(decoratedTransaction.value.toString()) : undefined,
+                gas: decoratedTransaction.gasLimit ? BigInt(decoratedTransaction.gasLimit.toString()) : undefined,
```
Comment (afeight) [inline packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:139]: all the properties on `decoratedTransaction` are potentially promises–should we be `await`ing them all?

Comment (afeight) [reply packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts:139]: does this work? Got a chatGPT answer but haven't tested:
```
async sendTransaction(transaction: Deferrable<TransactionRequest>): Promise<Hash> {
    try {
        const decoratedTransaction = await decorateSendTransactionData(transaction);
        const { to, value, gasLimit, nonce, data, maxFeePerGas, maxPriorityFeePerGas } = await resolveDeferrable(
            decoratedTransaction
        );

        return await this.kernelClient.sendTransaction({
            to: to as `0x${string}`,
            value: value ? BigInt(value.toString()) : undefined,
            gas: gasLimit ? BigInt(gasLimit.toString()) : undefined,
            nonce: await getNonce(nonce),
            data: await convertData(data),
            maxFeePerGas: maxFeePerGas ? BigInt(maxFeePerGas.toString()) : undefined,
            maxPriorityFeePerGas: maxPriorityFeePerGas ? BigInt(maxPriorityFeePerGas.toString()) : undefined,
            maxFeePerBlobGas: undefined,
            blobs: undefined,
            blobVersionedHashes: undefined,
            kzg: undefined,
            sidecars: undefined,
            type: undefined,
            chain: null,
        });
    } catch (error) {
        logError("[SEND_TRANSACTION] - ERROR_SENDING_TRANSACTION", {
            service: SCW_SERVICE,
            error: errorToJSON(error),
            transaction,
        });
        throw new TransactionError(`Error sending transaction: ${error}`);
    }
}

// let's put this in a helper file with `Deferrable`
async function resolveDeferrable<T>(obj: Deferrable<T>): Promise<T> {
    const resolved = {} as T;

    const entries = Object.entries(obj) as [keyof T, Promise<T[keyof T]>][];
    Promise.all(
        entries.map(async ([key, value]) => {
            resolved[key] = await value;
        })
    );

    return resolved;
}
```

Comment (lluiserena) [reply packages/client/wallets/aa/src/blockchain/wallets/EVMAAWallet.ts]: Added!

**Review by afeight — APPROVED** [review]
thanks!

**Pull request merged**